.github/workflows/nodejs.yml

```diff
@@ -138,10 +136,12 @@ jobs:
         run: npm ci
 
       - run: npm run build
-      - run: npm run test:integration
-        env:
-          HOST: localhost
-          PORT: ${{ job.services.rippled.ports['6006'] }}
+
+      # Docker stop must be in same step as integration test so it doesn't get skipped if tests fail.
```

Comment:
I'd recommend putting it in a separate step with `if: always()`

Reply:
Huh, I didn't know about conditionals - found this reference comment while googling, in this case I think `always()` is correct because by the time people cancel the test, I think the docker container will likely have already started, but in general apparently `!canceled()` is usually what people want.
https://github.com/actions/runner/issues/491#issuecomment-926924523